sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,72 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
-
+    typedef Song_t value_type;
+    typedef Song_t& reference;
+    typedef const Song_t& const_reference;
+    typedef typename Container::iterator iterator;
+    typedef typename Container::const_iterator const_iterator;
+    typedef ptrdiff_t difference_type;
+    typedef size_t size_type;
     /** @todo Iterators */
+    const_iterator begin() const{
+        return m_tracklist.cbegin();
+    }
+    const_iterator end() const{
+        return m_tracklist.cend();
+    }
 
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
-
+    template<class T>
+    StaticPlaylist(const T _container){
```

Comment:
You're copying the container, please don't! Container is a value type, so pass it by reference please
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const